js/donations-visualiser.js

```diff
@@ -855,49 +855,47 @@ function update(partyNodes, parties, selectedParties, resetControls) {
 
     d3.select("#party_select").selectAll(".checkbox").remove();
 
-    parties_grouped = []
-    parties_grouped_search = ["Coalition", "Australian Labor Party", "Australian Greens", "Palmer United Party"]
-    parties.map(function(item){
-      var index = parties_grouped_search.indexOf(party_map[item].name);
+    var major_party_names = ["Coalition", "Australian Labor Party", "Australian Greens", "Palmer United Party"];
+    // party indexes
+    var major_parties = [-1, -1, -1, -1];
```

Comment:
While the code works this way I worry that it's more brittle. For example, try adding "Australian Sex Party" and go to 2010 - 2011 on the filter. It throw an error, because the major parties array is [75, 18, 16, 4: 23]. This is because of the way I set the indexes in the forEach (setting indexes before I know the array is large enough to take them). Any ideas how to handle this properly?